### PR TITLE
Fix performance regression with `stddev` being enabled by default

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Related to https://github.com/apache/datafusion-comet/issues/824

## Rationale for this change

Fix a performance regression and simplify configs for enabling operators and expressions

tpcds_allqueries

We now fall back to Spark for `stddev_sample` aggregates.

image

## What changes are included in this PR?

- `stddev` is now disabled by default. There was a recent regression related to configs that had enabled this by default, and this operation is much slower in DataFusion than in Spark.
- Remove `spark.comet.exec.all.enabled` and enable all operators by default. Each operator can be disabled individually be changing it's enabled config to false. These are all documented.

## How are these changes tested?

Existing tests

### CODE REVIEW

File: native/core/src/execution/datafusion/planner.rs [hunk 1]
```diff
@@ -1208,17 +1208,17 @@ impl PhysicalPlanner {
...
+        fn prep_join_input(plan: Arc<dyn ExecutionPlan>) -> Arc<dyn ExecutionPlan> {
+            if can_reuse_input_batch(&plan) {
+                Arc::new(CopyExec::new(plan, CopyMode::UnpackOrDeepCopy))
+            } else {
+                Arc::new(CopyExec::new(plan, CopyMode::UnpackOrClone))
+            }
+        }
+
+        let left = prep_join_input(left);
+        let right = prep_join_input(right);
```
Comment (andygrove) [inline native/core/src/execution/datafusion/planner.rs:1221]: code cleanup: no functional change

**Comment by andygrove** [comment]
@parthchandra Could I get your opinion on the config changes in this PR? I'm not sure this is the best approach but it was previously not possible to have a default value of `false` for `spark.comet.exec.OPNAME.enabled` because it would be enabled anyway if all execs were enabled.

**Comment by parthchandra** [comment]
> I'm not sure this is the best approach but it was previously not possible to have a default value of `false` for `spark.comet.exec.OPNAME.enabled` because it would be enabled anyway if all execs were enabled.

Would it be better to have have an exclusion list (does not have to be a config, could just be an internal list) which is explicitly checked if EXEC_ALL is enabled?
So op is enabled if  ( (EXEC_ALL && not in EXEC_ALL_EXCLUSIONS) || op is enabled) ? 
We can include an op in exclusions if we feel it is not ready and it can be enabled explicitly for development/testing.

**Comment by andygrove** [comment]
There is an upstream PR in DataFusion to improve stddev performance
https://github.com/apache/datafusion/pull/12095

**Comment by andygrove** [comment]
I am closing this issue because we can disable this expression via a config now that https://github.com/apache/datafusion-comet/pull/855 is merged

**Pull request closed without merging**